app/src/main/java/com/swent/assos/ui/screens/assoDetails/AssoDetails.kt

```diff
@@ -160,17 +199,71 @@ fun TopAssoBar(
             modifier = Modifier.testTag("GoBackButton").clickable { navigationActions.goBack() })
       },
       actions = {
-        Button(
+        /*Button(
             modifier = Modifier.testTag("FollowButton"),
+            onClick = {
+                if (currentUser.following.contains(assoId)) viewModel.unfollowAssociation(assoId)
+                else viewModel.followAssociation(assoId)
+            },
+            colors = ButtonDefaults.buttonColors(containerColor = Color.Red)) {
+            Text(
+                modifier = Modifier.testTag("TextFollowButton"),
+                text = if (currentUser.following.contains(assoId)) "Unfollow" else "Follow",
+                color = Color.White)
+        }*/
+        AssistChip(
+            colors =
+                if (currentUser.following.contains(assoId))
+                    AssistChipDefaults.assistChipColors(containerColor = Color(0xFFD9D9D9))
+                else AssistChipDefaults.assistChipColors(containerColor = Color(0xFF4285F4)),
+            border = null,
+            modifier = Modifier.testTag("FollowButton").padding(5.dp),
             onClick = {
               if (currentUser.following.contains(assoId)) viewModel.unfollowAssociation(assoId)
               else viewModel.followAssociation(assoId)
             },
-            colors = ButtonDefaults.buttonColors(containerColor = Color.Red)) {
-              Text(
-                  modifier = Modifier.testTag("TextFollowButton"),
-                  text = if (currentUser.following.contains(assoId)) "Unfollow" else "Follow",
-                  color = Color.White)
-            }
+            label = {
+              if (currentUser.following.contains(assoId)) {
+                Text(
+                    text = "Following",
+                    color = Color.Black,
+                    fontFamily = FontFamily(Font(R.font.sf_pro_display_regular)),
+                    fontWeight = FontWeight.Medium,
+                )
+              } else {
+                Text(
+                    text = "Follow",
+                    color = Color.White,
+                    fontFamily = FontFamily(Font(R.font.sf_pro_display_regular)),
+                    fontWeight = FontWeight.Medium,
```

Comment:
To avoid code repetition, you could write only one text block and have an if condition on the text string and the color.